### DISCUSSION TITLE
Issue templates for maintainers only

### DISCUSSION THREAD

**Discussion by llvm-beanz** [community/community#62395] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

With all the great features that have been introduced to GitHub Projects (and more on the way), my team has been leaning more and more into using GitHub issues for tracking all our work. One of the challenges we’re having is around Issue templates. We’d really like to have templates for all the types of issues we need to file regularly. The problem is that we don’t want 10+ templates visible to our users who might be coming just to file a bug report or feature request.

It would be awesome if there was a way to configure the Issue template chooser to restrict which templates a user sees when filing an issue based on their role in the project. For example, users with no access see basic templates for filing bugs, feature requests and user support, but users with write access would also see templates for work items, release checklists, and other areas that only maintainers should really be poking at.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by talSofer** [comment]
I need this one too. Any feedback on it? 
@llvm-beanz were you able to workaround it in the meantime?

**Comment by jolelievre** [comment]
Hi, same here. This would be a great feature to differentiate internal team tasks from the community reports

**Comment by GuySartorelli** [comment]
We have a template that says something along the lines of "this template is for maintainers only, don't use this if you're not a maintainer" - but having that enforced would be fantastic.

**Comment by dcdenu4** [comment]
Plus one for wanting a release checklist template that is only visible to maintainers, a team, or organization.

**Comment by nightowl286** [comment]
+1 for this, or at least for allowing maintainers to create blank issues without a template, while having `blank_issues_enabled: false` for the public.
